aws_lambda_powertools/tracing/tracer.py

```diff
@@ -199,7 +225,11 @@ def put_annotation(self, key: str, value: Union[str, numbers.Number, bool]):
             return
 
         logger.debug(f"Annotating on key '{key}' with '{value}'")
-        self.provider.put_annotation(key=key, value=value)
+
+        if self._is_custom_provider():
+            self.provider.set_attribute(key=key, value=value)
+        else:
+            self.provider.put_annotation(key=key, value=value)  # type: ignore
```

Comment:
What if you just use `set_attribute` and make sure it works with both annotations and metadata on the Xray provider?

Reply:
This is to make sure the Tracer works the same way as it works before on X-Ray provider. We have considered unifying the behavior using `set_attribute`. 

To do so, firstly we need to make sure all the existing code calling `Tracer.put_annotation` is still calling `put_annotation` in SDK. But we didn't figure out a good way to do that. 

Secondly, current unit test is testing extensively if `provider.put_annotation`/`provider.put_metadata` is called. We need to change a lot of test cases to change `put_annotation` call `provider.set_attribute` instead. I think we will need an additional discussion to signoff the final design on this.